### PR TITLE
feat: server folder structure

### LINKED ISSUE #4521: [Timebox] feat: refactor server folder structure and modules
The goal of this ticket is to refactor the server folder structure, by separating business logic from the "core" now called engine of Twenty.
It will be easier for contributors to add features on twenty after that.
Also, in this ticket we'll refactor the modules structure as well.

### PR DESCRIPTION
**Pull request by magrinj**
This PR is finishing folder structuration of the `twenty-server`.
Maybe it's going to move again later, but this should add more clarity between nodules.

Fix #4521

### CODE REVIEW

File: packages/twenty-server/src/app.module.ts [hunk 1]
```diff
@@ -1,40 +1,44 @@
...
-    //   http: process.env.NODE_ENV !== 'production',
-    // }),
+    // Nest.js devtools, use devtools.nestjs.com to debug
+    DevtoolsModule.registerAsync({
+      useFactory: (environmentService: EnvironmentService) => ({
+        http: environmentService.get('DEBUG_MODE'),
+        port: environmentService.get('DEBUG_PORT'),
+      }),
+      inject: [EnvironmentService],
+    }),
```
Comment (magrinj) [inline packages/twenty-server/src/app.module.ts:27]: This is only enabled in debug mode

**Review by charlesBochet — APPROVED** [review]
Congratulations @magrinj , let's merge it as it is and iterate from there to unblock other workstream.
The folder structure looks great

However, I'm facing issues in local while testing your branch. Metadata cannot be retrieved and the FE crashes

**Review by Weiko — COMMENTED** [review]
Same thing happened, it seems some files were created as they could simply be renamed so we potentially lose history 🤔 Otherwise LGTM! (haven't tested it locally though)

File: packages/twenty-server/src/engine/core-modules/core-engine.module.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    AnalyticsModule,
+    AuthModule,
+    FeatureFlagModule,
+    TimelineMessagingModule,
+    TimelineCalendarEventModule,
+    UserModule,
+    WorkspaceModule,
+  ],
+})
+export class CoreEngineModule {}
```
[the remark this answers is not in the document]
Comment (magrinj) [reply packages/twenty-server/src/engine/core-modules/core-engine.module.ts:44]: Didn't like the repetition of `ModulesModule`, so I think it's clear enough that way but we can change it

**Comment by magrinj** [comment]
> Same thing happened, it seems some files were created as they could simply be renamed so we potentially lose history 🤔 Otherwise LGTM! (haven't tested it locally though)

When I make the change I only move the files, it's a git issue I guess, as it think it's a delete and create

File: packages/twenty-server/src/engine/core-modules/core-engine.module.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Weiko) [reply packages/twenty-server/src/engine/core-modules/core-engine.module.ts:44]: I agree ModulesModule was bad 😄 I'm more thinking of either renaming the fodler core-engine or the module CoreModule?

[the remark this answers is not in the document]
Comment (Weiko) [reply packages/twenty-server/src/engine/core-modules/core-engine.module.ts:44]: I see, lgtm 👍

**Pull request merged**